LDACheck_key_phrases.py

```diff
@@ -0,0 +1,199 @@
+	parser.add_argument("--lemmatizer", default=0, type=int) # for using lemmatization_tokenizer
+	parser.add_argument("--mx_ngram", default=2, type=int) # the upper bound of the ngram range
+	parser.add_argument("--mn_ngram", default=1, type=int) # the lower bound of the ngram range
+	parser.add_argument("--stop_words", default=1, type=int) # filtering out English stop-words
+	parser.add_argument("--vec_size", default=100, type=int) # the size of the vector in the semantics space
+	parser.add_argument("--min_count", default=12	, type=int) # minimum frequency of the token to be included in the vocabulary
+	parser.add_argument("--max_df", default=0.95, type=float) # how much vocabulary percent to keep at max based on frequency
+	parser.add_argument("--debug", default=0, type=int) # embed IPython to use the decomposed matrix while running
+	parser.add_argument("--compress", default="json", type=str) # for dumping resulted files
+	parser.add_argument("--out_dir", default="resultsTest", type=str) # name of the output directory
+	parser.add_argument("--min_nb_clusters", default=2, type=int) # minimum number of cluster we try
+	parser.add_argument("--max_nb_clusters", default=60, type=int) #  maximum number of cluster we try
+	parser.add_argument("--key_phrase", default="mental rotation", type=str) # the key phrase to retrieve in the metadata of the selected istex 
+
+	args = parser.parse_args()
+	input_file = args.input_file
+	output_file = args.output_file
+	out_dir = args.out_dir
+	
+	if not os.path.exists(out_dir):
+		os.makedirs(out_dir)
+	lemmatizer = args.lemmatizer
+	min_nb_clusters = args.min_nb_clusters
+	max_nb_clusters = args.max_nb_clusters
+	key_phrase = args.key_phrase
+
+	if lemmatizer:
+		lemmatizer = Lemmatizer()
+	else:
+		lemmatizer = None
+	mx_ngram = args.mx_ngram
+	mn_ngram =  args.mn_ngram
+	stop_words = args.stop_words
+	if stop_words:
+		stop_words = 'english'
+	else:
+		stop_words = None
+	n_components = args.vec_size
+	min_count = args.min_count
+	max_df = args.max_df
+	debug = args.debug
+	compress = args.compress
+	out_dir = args.out_dir
+
+	# instead of recomputing the vectors, we should use the one of the complete experiment, so use  pickle load
+	f = open(input_file, "r")
+	input_dict = pickle.load(f)
+	nb_articles = len(input_dict)
+	f.close()
+
+	keys, values = KeysValuesInit(nb_articles)
+	nb_random_with_key_phrase = stat_check_vocabulary(keys, values, groups_avoid=["UCBL", "MRISTEX"], key_phrase=key_phrase)
+	tf_idf_vectorizer = TfidfVectorizer(input='content', analyzer='word', stop_words=stop_words, tokenizer=lemmatizer,
+			min_df=min_count,  ngram_range=(mn_ngram, mx_ngram), max_df=max_df)
+
+	tf_idf_bow = tf_idf_vectorizer.fit_transform(values)
+	tf_feature_names = tf_idf_vectorizer.get_feature_names()
+
+	generic = open(output_file, "w")
+	ucbl_out = open(os.path.join(out_dir, "lda_ucbl_cluster.txt"), "w")
+	istex_out = open(os.path.join(out_dir, "lda_mristex_cluster.txt"), "w") 
```

Comment:
Instead of using 'istex_out' as a global variable to used in the function 'statiticsClusters', you should pass the output_file as a function parameter and move this open file line to the function